tools/state-parts-dump-check/src/cli.rs

```diff
@@ -617,6 +726,44 @@ async fn process_part_with_3_retries(
     res?
 }
 
+async fn process_header_with_3_retries(
+    chain_id: String,
+    epoch_id: EpochId,
+    epoch_height: u64,
+    shard_id: ShardId,
+    external: ExternalConnection,
+) -> anyhow::Result<()> {
+    let mut retries = 0;
+    let mut res: Result<Result<(), anyhow::Error>, tokio::time::error::Elapsed>;
+    loop {
+        let chain_id = chain_id.clone();
+        let epoch_id = epoch_id.clone();
+        let external = external.clone();
+        let timeout_duration = tokio::time::Duration::from_secs(60);
+        res = timeout(
+            timeout_duration,
+            process_header(chain_id, epoch_id, epoch_height, shard_id, external),
+        )
+        .await;
+        match res {
+            Ok(Ok(_)) => {
+                tracing::info!(shard_id, epoch_height, "process_header success.",);
+                break;
+            }
+            _ if retries < MAX_RETRIES => {
+                tracing::info!(shard_id, epoch_height, "process_header failed. Will retry.",);
```

Comment:
maybe would be helpful to log the error here and also the match arm below

Reply:
Done